sphinx/texinputs/sphinxlatexshadowbox.sty

```diff
@@ -113,4 +148,31 @@
    \spewnotes
   }
 
+% 8.1.0
+\newenvironment{sphinxtopic}
+               {\begin{sphinxShadowBox}\relax}{\end{sphinxShadowBox}}
```

Comment:
The environment now admis an optional argument.  I put the `\relax` to be extra sure some `[` will not fool it. However you are right it can not happen for topic directive as it requires a title so Sphinx will always output stuff like this
```latex
\begin{sphinxtopic}
\sphinxstyletopictitle{A}
```
and there can't be a problem.

Yes using `[topic]` explictly is clearer for people reading this `.sty` file but it is probably slightly less efficient, now that doesn't matter.  If people want to use directly the environment they have to be cautious though with the `[` trap, so adding the `[topic]` here would help them avoid mistakes.

+0 for using ``[topic]`` will wait my 12 hours of usual sleep for a decision :smiley:

Reply:
After some closer look, and for reasons already mentioned, the `\relax` is not needed indeed. Besides Sphinx escapes `[` to `{[}` to avoid potential problems (mainly in lists).  Then I hesitated with using the explicit `[topic]` or nothing at all, and opted for nothing at all, in the end.  No strong opinion.

Reply:
I personally prefer the explicit `topic` because the default was only kept for backwards compatibility. In the future, we might at some point remove it or change it for whatever reasons. How do you feel about this?

Reply:
ok, agreed. I have pushed with `[topic]` added.

Reply:
Oh thank you for this! One thing I always found extremely tricky in LaTeX (and TeX) is about those space tokens and how they are dealt with... So it helps a lot when you explain it!

Reply:
About my example, LaTeX provides `\ignorespacesafterend` which you can use when defining the `end` part of an environment but no `\ignorespacesatbegin` which user could put in the `begin` part.

About what you say about spaces I fully agree.  There is a blatant lack of a serious document "TeX/LaTeX spaces explained". With the promotion of LaTeX3 with whom programmers do not have to worry at all anymore about spaces in their source code (they get ignored by default), I am sure situation will degrade because the folks doing LaTeX "programming" have less incentives to care about spaces...